tests/foreman/api/test_discoveredhost.py

```diff
@@ -356,6 +393,21 @@ def test_positive_reboot_pxe_host(self):
 
         :CaseImportance: Medium
         """
+        sat = module_discovery_sat.sat
+        provisioning_host.power_control(ensure=False)
+        mac = provisioning_host._broker_args['provisioning_nic_mac_addr']
+        wait_for(
+            lambda: sat.api.DiscoveredHost().search(query={'mac': mac}) != [],
+            timeout=240,
+            delay=20,
+        )
+        discovered_host = sat.api.DiscoveredHost().search(query={'mac': mac})[0]
+        discovered_host.hostgroup = provisioning_hostgroup
+        discovered_host.location = provisioning_hostgroup.location[0]
+        discovered_host.organization = provisioning_hostgroup.organization[0]
+        discovered_host.build = True
+        result = sat.api.DiscoveredHost(id=discovered_host.id).reboot()
+        assert 'Unable to perform reboot' not in result
```

Comment:
Shouldn't we actually check if the host has successfully rebooted?

Reply:
Yeah so after the reboot we have all the relevant host information that we can assert like the name/org/location, but did not find a good candidate to assert for the reboot there, hence a negative assertion makes things more straightforward.